hack/tools/release/notes/main.go

```diff
@@ -179,7 +182,11 @@ func computeConfigDefaults(config *notesCmdConfig) error {
 	}
 
 	if config.toRef == "" {
-		config.toRef = "heads/" + config.branch
+		if newTag.Patch == 0 {
+			config.toRef = release.TagsPrefix + fmt.Sprintf("v%d.%d.0", newTag.Major, newTag.Minor)
+		} else {
+			config.toRef = release.TagsPrefix + fmt.Sprintf("v%d.%d.%d", newTag.Major, newTag.Minor, newTag.Patch)
+		}
```

Comment:
Apologies, I'm being obtuse here
Could you explain this logic?
If `toRef` is not set explicitly, we set the ref to the next tag? But tag doesn't exist at the time we generate these notes, so how does that work?

Reply:
Ah I think this was due to me trying to generate release notes for a previous version, and dependencies that were added on the branch long after the tag was created were being listed in the release notes. If we change this back to point to head, the unit tests would never pass. The correct thing to do would be to revert this change and point it back to head, but the issue with that is we won't be able to generate release notes accurately for previous tags. Maybe we can restructure the unit test?